### ISSUE TITLE
Error using factions 2.7.5

### ISSUE THREAD

**Issue by tehrenderr** [MarkehMe/FactionsFramework#1]
http://pastebin.com/LRSyqYAG

**Comment by tehrenderr** [comment]
Have you looked into it?

**Comment by MarkehMe** [comment]
I have! 😄  I need to change the support layers around. 

If you want to try it on on other 2.7 versions it might give me a better understanding, I just have a lot of work on at the moment - lacking time to work on it!

**Comment by MarkehMe** [comment]
@tehrenderr can you confirm your MassiveCore version please?

**Comment by tehrenderr** [comment]
Massivecore v2.7.5

**Comment by MarkehMe** [comment]
Fixed.

**Comment by MarkehMe** [comment]
🎉  **FactionsFramework v1.1.0 has been released and addresses this issue** 🎉 

You can grab it from:
* [GitHub](https://github.com/MarkehMe/FactionsFramework/releases/tag/1.1.0)
* [Spigot Resources](https://www.spigotmc.org/resources/factions-framework.22278/update?update=89455)
* [BukkitDev](http://dev.bukkit.org/bukkit-plugins/factionsframework/files/2-factions-framework-v1-1-0/)

```
v1.1.0 Changelog:

* Factions#getNone() has been deprecated and will be unavailable in 6 months
* Factions#getNone(World) added
* Factions#getSafezone(World) added
* Factions#getWarzone(World) added
* Stability fixes for 2.8.6, 2.8.2, 2.7, 2.6
* Supports FactionsOne (1.8)
* Uses a gradle module layout (less reflection now too!)
* Internal changes and performance enhancements
```

**Comment by tehrenderr** [comment]
Still doesn't work -_-
http://pastebin.com/xiyzz4cA

**Comment by MarkehMe** [comment]
Updating FactionsFramework as this was a bug inside FactionsFramework, not FactionsWarps.

### FactionsFramework
You can download it from Spigot Resources, BukkitDev, or from the [GitHub page](https://github.com/MarkehMe/FactionsFramework/releases).

[![Spigot Resources](https://www.spigotmc.org/favicon.ico "Spigot Resources") Spigot Resources](https://www.spigotmc.org/resources/factions-framework.22278/history)

[![BukkitDev](http://dev.bukkit.org/media/site-favicons/0/7/favicon.ico "BukkitDev") BukkitDev](http://dev.bukkit.org/bukkit-plugins/factionsframework/files/)

**Comment by MarkehMe** [comment]
Looking at your error log makes me have a bit of disbelief here.. it says Factions_2_8_2, yet when I test I get 2_7. Can you confirm you actually upgraded the plugin? 

Are you using a modified plugin? 

You can see this on your console when you start the server:

```
[15:06:22 INFO]: [FactionsFramework] Enabling FactionsFramework v1.1.0
[15:06:23 INFO]: [FactionsFramework] Factions version is: Factions_2_7
[15:06:23 INFO]: [FactionsWarps] Enabling FactionsWarps v1.1.0
[15:06:23 INFO]: Done (14.370s)! For help, type "help" or "?"
```

**Comment by tehrenderr** [comment]
My factions is custom coded and It is 2.7.5 factions but it may have been compiled using 2.8 code or something like that. I can send you the source for my factions and massivecore, but I definitely upgraded factionsframework.

Here is the source:
https://www.dropbox.com/s/9djhepkzep8sibx/Factions%202.7.5%20SRC%20PLus%20MassiveCore%20SRC.zip?dl=0

**Comment by MarkehMe** [comment]
Do you have a change log, or have this on a git repo page where I can compare the changes? 

Is this a public plugin?

**Comment by tehrenderr** [comment]
It is a private plugin, really nothing was changed besides adding on features so I don't understand how this would cause factionsframework to break.

**Comment by MarkehMe** [comment]
If you try it with the regular Factions and Massive core version 2.7.5 (attached) it will work fine. 

So something has changed internally from the original code. What features have been added (or removed) - in this case it looks like ```com.massivecraft.massivecore.MassiveException``` was added to the class path, whereas 2.7.5 doesn't have this. 

This is hard because the 2.8.x commands require throwing MassiveException, but 2.7.5 commands aren't able to. Your alternate version seems to want the best of both worlds. 

Is it possible for your developer to fix the MassiveCore version? As this is as if you were using Factions 2.7.5 and MassiveCore 2.8.x? 

_Ref: [Factions and MassiveCore version 2.7.5](https://github.com/MarkehMe/FactionsFramework/files/258608/Factions_Massivecore_2.7.5.zip)_

**Comment by MarkehMe** [comment]
Just to confirm.. MassiveCore 2.7.6 does not have ```com.massivecraft.massivecore.MassiveException```

![](http://i.imgur.com/Kke13ql.png)

**Comment by tehrenderr** [comment]
I just contacted my dev about this issue, thanks for looking into this. I'll get back to you once he does.

**Comment by MarkehMe** [comment]
Keep me posted. 😃

**Comment by grrocks** [comment]
Hmm when I downloaded the SRC of MassiveCore I used this commit: https://github.com/MassiveCraft/MassiveCore/tree/ff8a68b82eb8544039959543af1d7738d6919a0e and this on is the last commit before 2.8.0

**Comment by MarkehMe** [comment]
If you look in the source I was just provided it contains **MassiveException.java** - this is the deciding factor for the 2_7 layer

 _../src/com/massivecraft/massivecore/_
```ruby
$ ls | grep Massive
MassiveCore.java
MassiveCoreEngineCommandRegistration.java
MassiveCoreEngineDestination.java
MassiveCoreEngineMain.java
MassiveCoreEngineVariable.java
MassiveCoreEngineWorldNameSet.java
MassiveCoreMConf.java
MassiveCoreMConfColl.java
MassiveCorePerm.java
MassiveCoreTaskDeleteFiles.java
MassiveException.java # <------ MassiveException is included
MassivePlugin.java

```

**Comment by MarkehMe** [comment]
Just did  quick look for 2.7.0 - 2.7.4 as well, none include MassiveException:

* [2.7.4](https://github.com/MassiveCraft/MassiveCore/tree/d09d2b4169f08e51d6bf94b2cc1f3e7e4672dcab/src/main/java/com/massivecraft/massivecore)
* [2.7.3](https://github.com/MassiveCraft/MassiveCore/tree/8777b9678f951d003dbc55f0ae9e53d48ff3e9ce/src/main/java/com/massivecraft/massivecore)
* [2.7.2](https://github.com/MassiveCraft/MassiveCore/tree/b703054ccb305debfab6f182f0656d4d02e9cc6d/src/main/java/com/massivecraft/massivecore)
* [2.7.1](https://github.com/MassiveCraft/MassiveCore/tree/6aa2bd5ee41d349eafa703878e64b14f93f6e5fe/src/main/java/com/massivecraft/massivecore)
* [2.7.0](https://github.com/MassiveCraft/MassiveCore/tree/284debd11c10cc69d2abeb22efc16a0ececc115d/src/main/java/com/massivecraft/massivecore)

**Comment by grrocks** [comment]
What decides the layer for the Factions plugin?

**Comment by MarkehMe** [comment]
2.8.x plugins throw ```com.massivecraft.massivecore.MassiveException``` in their FactionsCommand object (see: https://github.com/MarkehMe/FactionsFramework/blob/master/layers/Layer_2_8_2/src/main/java/me/markeh/factionsframework/layer/layer_2_8_2/Command_2_8_2.java#L55 for an example)

2.7.x FactionsCommand does not use it (see https://github.com/MarkehMe/FactionsFramework/blob/master/layers/Layer_2_7/src/main/java/me/markeh/factionsframework/layer/layer_2_7/Command_2_7.java#L44 for an example) 

You can fix your issue by just using the correct version of MassiveCore @ https://github.com/MassiveCraft/MassiveCore/tree/23d11564244e7363fa187e61108fad2da1d26d7f - is there a reason you can't do this?

**Comment by grrocks** [comment]
The issue is I would have to use the released 2.7.5 Factions to make is compatible with that commit. but I guess that is the easiest way to fix this issue I guess this will be the option I will go through, Thanks!
